src/plugins/displayLayout/components/TelemetryView.vue

```diff
@@ -169,11 +169,30 @@ export default {
                 return;
             }
 
-            if (this.item.format) {
-                return printj.sprintf(this.item.format, this.datum[this.valueMetadata.key]);
+            const itemFormat = this.item.format;
```

Comment:
I think there's an opportunity for cleanup of the original code, which will make migrating this to VISTA easier (less duplication of code), and will have the side benefit of improving performance.

We should implement a new `CustomStringFormatter` telemetry formatter, which takes a string format as an argument to its constructor.

eg. 
```
const format = new CustomStringFormat(this.item.format);
```
All of the logic for custom formatting can be moved in there (for both printf style formatting and for invoking Open MCT custom formatters). This will clean up the code a little bit and save us having to re-implement it everywhere we want to do custom formatting.

To improve performance we should only create this formatter when the underlying format changes. Therefore, we should observe `this.item.format` and set the formatter as a view model property. (btw, why is the domain object referenced as `this.item`? A mystery for the ages...)

`telemetryValue` would still be computed property but since the formatter would only change when the user changes the custom format Vue will be able to cache results which should improve performance slightly.